modules/apps/frontend-theme/frontend-theme-classic/src/css/custom_properties/_custom_properties_set.scss

```diff
@@ -50,79 +50,130 @@ body {
 
 // Buttons
 
-.btn-link:not(.dropdown-item) {
-	@include button-cp(
-		var(--transparent),
-		var(--transparent),
-		var(--primary),
-		var(--transparent),
-		var(--transparent),
-		var(--primary-d1)
-	);
-}
+.btn:not(.btn-link) {
+	position: relative;
+	z-index: 1;
+
+	&::after {
+		background-color: black;
+		border-radius: clay-enable-rounded($btn-border-radius);
+		bottom: math-sign($btn-border-width);
+		content: '';
+		left: math-sign($btn-border-width);
+		opacity: 0;
+		position: absolute;
+		right: math-sign($btn-border-width);
+		top: math-sign($btn-border-width);
+		transition: opacity 150ms ease-in-out;
+		will-change: opacity;
+		z-index: -1;
+	}
 
-.btn-outline-light {
-	@include button-cp(
-		var(--transparent),
-		var(--white),
-		var(--white),
-		var(--white),
-		var(--white),
-		var(--dark)
-	);
-}
+	&:hover,
+	&:focus {
+		&::after {
+			opacity: 0.2;
+		}
+	}
 
-.btn-outline-primary {
-	@include button-cp(
-		var(--transparent),
-		var(--primary),
-		var(--primary),
-		var(--primary-d1),
-		var(--primary-d1),
-		var(--white)
-	);
-}
+	&:active,
+	&.active {
+		&::after {
+			opacity: 0.3;
+		}
+	}
 
-.btn-outline-secondary {
-	@include button-cp(
-		var(--transparent),
-		var(--gray-400),
-		var(--secondary),
-		var(--gray-200),
-		var(--transparent),
-		var(--dark)
-	);
+	&.disabled,
+	&:disabled {
+		&::after {
+			opacity: 0;
+		}
+	}
 }
 
-.btn-primary {
-	@include button-cp(
-		var(--primary),
-		var(--primary),
-		var(--white),
-		var(--primary-d1),
-		var(--primary-d1)
+.btn-link:not(.dropdown-item) {
+	@include clay-button-variant(
+		(
+			background-color: var(--transparent),
+			border-color: var(--transparent),
+			color: var(--primary),
+			hover: (
+				background-color: var(--transparent),
+				border-color: var(--transparent),
+				color: var(--primary),
+			),
+			focus: (
+				background-color: var(--transparent),
+				border-color: var(--transparent),
+				color: var(--primary),
+			),
+			active: (
+				background-color: var(--transparent),
+				border-color: var(--transparent),
+				color: var(--primary),
+			),
+		)
 	);
 }
 
-.btn-secondary {
-	@include button-cp(
-		var(--white),
-		var(--gray-400),
-		var(--secondary),
-		var(--gray-100),
-		var(--gray-400),
-		var(--dark)
-	);
-}
+@each $key, $value in $custom-properties-theme-colors {
+	$background-color: $value;
+	$color: var(--white);
 
-.btn-success {
-	@include button-cp(
-		var(--success),
-		var(--success),
-		var(--white),
-		var(--success-d1),
-		var(--success-d1)
-	);
+	@if ($key == 'secondary') {
+		$background-color: var(--white);
+		$color: $value;
+	}
+
+	.btn-#{$key} {
+		@include clay-button-variant(
+			(
+				background-color: #{$background-color},
+				border-color: #{$value},
```

Comment:
Instead of mapping each variant I think we can use the defined theme-colors and add the exception for secondary to loop.

My only concern is that we currently have gray-dark and lighter as theme-colors and we are generating also those buttons

<img width="406" alt="Screenshot 2020-08-27 at 11 28 31" src="https://user-images.githubusercontent.com/46778114/91428919-dbc65700-e85e-11ea-8fd1-a32f2b173993.png">

<img width="1440" alt="Screenshot 2020-08-27 at 12 00 35" src="https://user-images.githubusercontent.com/46778114/91429082-e680ec00-e85e-11ea-9f16-92b9e3965e1a.png">

But these new buttons depends on the gray-800 and gray-100 colors, so they are still customizable

Reply:
How close are we trying to keep this to Clay CSS API? The way we're creating `$custom-properties-theme-colors` won't generate new theme colors if someone adds it via Clay CSS Sass variables. The same goes for `$custom-properties-spacers`.

Reply:
I don't think `gray-dark` and `lighter` needs to be a theme color if only using it to generate `.bg-gray-dark` and `.text-gray-dark`.
```
.bg-gray-dark { background-color: var(--gray-800); }
.text-gray-dark { color: var(--gray-800); }
.bg-lighter { background-color: var(--gray-100); }
.text-lighter { color: var(--gray-100); }
```
if you really want to generate it:
```
$custom-properties-background-text-colors: () !default;
$custom-properties-background-text-colors: map-merge(
  $custom-properties-theme-colors,
  $custom-properties-background-text-colors
);
$custom-properties-background-text-colors: map-merge((
  gray-dark: var(--gray-800),
  lighter: var(--gray-100),
), $custom-properties-background-text-colors);

@each $color, $value in $custom-properties-background-text-colors {
  .bg-#{$color} {
    background-color: $value !important;
  }

  .text-#{$color} {
    color: $value !important;
  }
}
```

I don't like using Sass loops to generate components. It generates a lot of CSS that we may or may not need.

Reply:
You are right but I'm not sure what to do because if I recall correctly @atjuan asked for these new theme colors for something he was building and normally I'd ask to him but he's on PTO this week so we are on our own 😬 – I think I'll go with your solution because it makes sense and if he need it we can always add those colors back